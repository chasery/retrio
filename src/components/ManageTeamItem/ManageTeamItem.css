.ManageTeamItem__role {
  margin-top: 0.5rem;
}
.ManageTeamItem__email {
  font-weight: normal;
}
.ManageTeamItem__controls > * {
  margin-right: 0.5rem;
}
