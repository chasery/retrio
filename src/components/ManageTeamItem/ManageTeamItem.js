import React, { useContext } from 'react';
import { useHistory, useParams } from 'react-router-dom';
import RetrioContext from '../../context/retrio-context';
import './ManageTeamItem.css';

function ManageTeamItem(props) {
  const { id, name, email } = props;
  const context = useContext(RetrioContext);
  const { teamId } = useParams();
  const history = useHistory();

  const team = context.teams.find((team) => team.id === teamId);

  const canRemove = () => {
    if (
      team
        ? team.owner_id
        : 1 === context.loggedInUser.id && id !== context.loggedInUser.id
    ) {
      return true;
    } else {
      return false;
    }
  };

  const canLeave = () => {
    if (
      team
        ? team.owner_id
        : 1 !== context.loggedInUser.id && context.loggedInUser.id === id
    ) {
      return true;
    } else {
      return false;
    }
  };

  const handleRemoveTeamMember = () => {
    context.removeTeamMember(teamId, id);
  };

  const handleLeaveTeam = () => {
    // Will remove relationship of user to team in further development.
    // For prototype purposes, we'll just remove the team to mimic
    // the behavior of not seeing the team any longer.
    context.deleteTeam(teamId);
    history.push(`/teams`);
  };

  return (
    <li className='ManageTeamItem'>
      <div className='ManageTeamItem__info'>
        <h3 className='ManageTeamItem__name'>{name}</h3>
        <div className='ManageTeamItem__email'>{email}</div>
        <div className='ManageTeamItem__role'>
          {team ? team.owner_id : 1 === id ? 'Owner' : 'Contributor'}
        </div>
      </div>

      {canRemove() && (
        <div className='ManageTeamItem__controls'>
          <button className='Link' onClick={handleRemoveTeamMember}>
            Remove
          </button>
        </div>
      )}
      {canLeave() && (
        <div className='ManageTeamItem__controls'>
          <button className='Link' onClick={handleLeaveTeam}>
            Leave
          </button>
        </div>
      )}
    </li>
  );
}

export default ManageTeamItem;
