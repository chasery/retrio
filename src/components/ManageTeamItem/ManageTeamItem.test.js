import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ManageTeamItem from './ManageTeamItem';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useParams: () => ({
    id: '',
  }),
}));

describe('ManageTeamItem component', () => {
  it('renders without error', () => {
    const wrapper = shallow(<ManageTeamItem />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
