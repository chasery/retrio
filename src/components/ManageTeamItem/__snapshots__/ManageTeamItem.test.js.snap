// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManageTeamItem component renders without error 1`] = `
<li
  className="ManageTeamItem"
>
  <div
    className="ManageTeamItem__info"
  >
    <h3
      className="ManageTeamItem__name"
    />
    <div
      className="ManageTeamItem__email"
    />
    <div
      className="ManageTeamItem__role"
    >
      Contributor
    </div>
  </div>
  <div
    className="ManageTeamItem__controls"
  >
    <button
      className="Link"
      onClick={[Function]}
    >
      Leave
    </button>
  </div>
</li>
`;
